src/ServiceManagement/Services/Commands.Utilities/Common/ServiceManagementBaseCmdlet.cs

```diff
@@ -215,11 +215,18 @@ protected OperationStatusResponse GetOperationNewSM(string operationId)
             {
                 try
                 {
-                    result = action();
+                    try
+                    {
+                        result = action();
+                    }
+                    catch (CloudException ce)
+                    {
+                        throw new ComputeCloudException(ce);
+                    }
                 }
-                catch (CloudException ex)
+                catch (Exception ex)
                 {
-                    WriteExceptionDetails(ex);
+                    WriteExceptionError(ex);
                 }
```

Comment:
This is expected - capturing the cloud exception and re-throwing the wrapped one. May I know why you think it should be avoided?